[12.0][MIG] stock_pack_operation_auto_fill > stock_move_line_auto_fill

jarroyomorales opened the pull request:
Migration to 12.0

zakmak88 committed:
[9.0][IMP]stock_pack_operation_auto_fill: fill operations automatically

Commit:
[MIG] stock_pack_operation_auto_fill: Migration to 10.0

zakmak88 committed:
[10.0][REF]stock_pack_operation_auto_fill: move auto fill button to header

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[MIG]stock_pack_operation_auto_fill: Migration to 11.0

Commit:
[FIX][stock_pack_operation_auto_fill] - Wrong qty done when use different UoM.

oca-transbot committed:
OCA Transbot updated translations from Transifex

sergio-teruel committed:
[11.0][IMP] stock_pack_operation_auto_fill: Merge module stock_picking_transfer_lot_autoassign

oca-travis committed:
[UPD] Update stock_pack_operation_auto_fill.pot

oca-travis committed:
[UPD] Update stock_pack_operation_auto_fill.pot

OCA-git-bot committed:
[UPD] README.rst

chienandalu committed:
[FIX] stock_pack_operation_auto_fill: Autofill button permissions

Only stock users should be able to process done quantities.

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

Commit:
[12.0][MIG] stock_pack_operation_auto_fill

rousseldenis commented:
Hi @jarroyomorales. Thanks for this.

As the model name has changed (and also the model description), it's better to rename it stock_move_line_auto_fill (with a migration script).

I saw the remark here :
 https://github.com/OCA/stock-logistics-workflow/issues/457

jarroyomorales commented:
Hi @rousseldenis ,
I just changed the name and the description and added the rename in OpenUpgrade.
https://github.com/OCA/OpenUpgrade/pull/1898
Is there anything else that needs to be done?

HaraldPanten reviewed (approved):
Functional review 👍

kittiu commented on stock_move_line_auto_fill/README.rst:
```diff
@@ -0,0 +1,143 @@
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module alow auto fill quantities in picking operations and autoassignment
```
```suggestion
This module allow auto fill quantities in picking operations and autoassignment
```

pedrobaeza replied:
s/allow/allows to

jarroyomorales replied:
Changed

jarroyomorales replied:
Changed

kittiu commented on stock_move_line_auto_fill/README.rst:
```diff
@@ -0,0 +1,143 @@
+This module alow auto fill quantities in picking operations and autoassignment
+lots quantities.
+
+In Odoo, if you schedule to transfer 50 product quantities and only receive 49
```
I don't quite understand this statement. It means receive (or transfer) 50 lines of product or 1 product with 50 units, etc.
@jarroyomorales I understand that the readme is interited from older version. But I do find a hard time trying to understand the readme, otherwise have to dig into the code to understand. Is it possible to enhance the overall readme files? :-)

jarroyomorales replied:
It means 50 lines of product, I don't know if the small change I did helps understand it better. By saying just 50 products instead of 50 product quantities I think its easier to understand it means lines.

kittiu commented on stock_move_line_auto_fill/README.rst:
```diff
@@ -0,0 +1,143 @@
+    * Pedro M. Baeza <pedro.baeza@tecnativa.com>
+    * Vicent Cubells <vicent.cubells@tecnativa.com>
+    * Sergio Teruel <sergio.teruel@tecnativa.com>
+    * David Vidal <david.vidal@tecnativa.com>
```
As person who migrate this addon, I believe you can add your credit here.

jarroyomorales replied:
OK!

kittiu commented on stock_move_line_auto_fill/models/stock_picking_type.py:
```diff
@@ -0,0 +1,16 @@
+
+    auto_fill_operation = fields.Boolean(
+        string="Auto fill operations",
+    )
```
help="To auto fill done quantity in picking document.\n- If checked, auto fill done quantity automatically\n- If unchecked, show button AutoFill for user to do the auto fill manually",

jarroyomorales replied:
Changed

kittiu commented on stock_move_line_auto_fill/models/stock_picking_type.py:
```diff
@@ -0,0 +1,16 @@
+    avoid_lot_assignment = fields.Boolean(
+        string="Avoid auto-assignment of lots",
+        default=True,
+    )
```
help="Avoid auto fill for more line with lots product",

kittiu replied:
I think it would be nice to have some help there ;)

jarroyomorales replied:
Changed

kittiu commented:
Functional tested LGTM

kittiu commented on stock_move_line_auto_fill/README.rst:
```diff
@@ -0,0 +1,143 @@
+that you only have to change it in case of divergence, but having the
+possibility of transferring directly.
+
+Also this module adds a button in backorder confirmation wizard to auto
```
Does this module actually have the feature specified here, I can't find new button in code? "A button in back order confirmatoin wizard to auto ... "

jarroyomorales replied:
Yes, that's true, I removed that paragraph from the README

bealdav committed:
IMP mv_line_auto_fill: pick type UI

bealdav commented on stock_move_line_auto_fill/views/stock_picking_type_views.xml:
```diff
@@ -0,0 +1,16 @@
+            <field name="use_existing_lots" position="after">
+                <field name="auto_fill_operation"/>
+                <field name="avoid_lot_assignment" groups="stock.group_production_lot"/>
+            </field>
```
Here is an improvement
https://github.com/tegin/stock-logistics-workflow/pull/1/files

jarroyomorales replied:
Thank you!

bealdav reviewed (commented):
An UI improvement as suggestion

jarroyomorales committed:
Merge pull request #1 from akretion/12-auto-fill-view

IMP mv_line_auto_fill: pick type UI

rousseldenis commented:
@bealdav @pedrobaeza @kittiu Could you update your review ?

bealdav reviewed (approved):
My suggestion merged.
Works in production

pedrobaeza commented:
/ocabot merge

rafaelbn commented:
We should clarify @sergio-teruel if this FIX is needed here in v12 https://github.com/OCA/stock-logistics-workflow/pull/546
